components/dashboard/src/components/start-workspace.tsx

```diff
@@ -158,15 +164,43 @@ export class StartWorkspace extends React.Component<StartWorkspaceProps, StartWo
     protected setErrorState(err: any, defaultErrorMessage?: string) {
         let errorMessage = defaultErrorMessage;
         let errorCode = undefined;
+        let errorData = undefined;
         if (err instanceof ResponseError) {
             errorCode = err.code;
+            errorData = err.data;
+            errorMessage = err.message;
+
+            switch (err.code) {
```

Comment:
Why not move the handling to "renderError"?

Reply:
1. (and foremost) I tried to align with `create-workspace.tsx` as much as possible and avoided adding unknown bits
2. here, it's mostly about redirecting right after getting an error, not so much about rendering error state.

Reply:
> (and foremost) I tried to align with create-workspace.tsx as much as possible and avoided adding unknown bits